pkg/nvpci/nvpci.go

```diff
@@ -339,9 +389,13 @@ func (p *nvpci) GetGPUByPciBusID(address string) (*NvidiaPCIDevice, error) {
 		NumaNode:   int(numaNode),
 		Config:     config,
 		Resources:  resources,
-		IsVF:       isVF,
 		DeviceName: deviceName,
 		ClassName:  className,
+		SriovInfo:  sriovInfo,
+	}
+	// cache physical functions only as VF can't be a root device.
+	if cache != nil && nvdevice.IsPF() {
+		cache[address] = nvdevice
 	}
```

Comment:
Let's move this above the construction of the NvidiaPCIDevice type.

Reply:
Then lets add IsPF() and IsVF() calls to the `SriovInfo` type (and then just call these from the `NvidiaPCIDevice` as appropriate.

Reply:
changed.